packages/ui/src/hooks/useUserCollections.ts

```diff
@@ -0,0 +1,72 @@
+import { paths, setParams } from '@reservoir0x/reservoir-sdk'
+import useReservoirClient from './useReservoirClient'
+import useSWRInfinite, { SWRInfiniteConfiguration } from 'swr/infinite'
+
+type UserCollections =
+  paths['/users/{user}/collections/v2']['get']['responses']['200']['schema']
+type UserCollectionsQuery =
+  paths['/users/{user}/collections/v2']['get']['parameters']['query']
+
+export default function (
+  user?: string,
+  options?: UserCollectionsQuery,
+  swrOptions: SWRInfiniteConfiguration = {}
+) {
+  const client = useReservoirClient()
+
+  const { data, mutate, error, isValidating, size, setSize } =
+    useSWRInfinite<UserCollections>(
+      (pageIndex, previousPageData) => {
+        if (!user) {
+          return null
+        }
+        const url = new URL(
+          `${client?.apiBase || ''}/users/${user}/collections/v2`
+        )
+        let query: UserCollectionsQuery = 
+        {
+          offset: pageIndex * 100, 
+          limit: 100,
```

Comment:
The limit seems a bit high, also if someone passes in a different limit our offset will be broken. We should make a variable and check if there's a limit passed to override the default, then make the offset dynamic based on that.